sockeye/training.py

```diff
@@ -85,15 +114,40 @@ def __init__(self,
                  context: List[mx.context.Context],
                  train_iter: data_io.ParallelBucketSentenceIter,
                  bucketing: bool,
-                 lr_scheduler) -> None:
+                 lr_scheduler,
+                 state_names: Optional[List[str]]) -> None:
         super().__init__(config)
+        self._get_sampling_threshold = self._get_sampling_scheduler()
         self.context = context
         self.lr_scheduler = lr_scheduler
         self.bucketing = bucketing
+        self.state_names = state_names if state_names is not None else []
         self._build_model_components()
         self.module = self._build_module(train_iter)
         self.training_monitor = None  # type: Optional[callback.TrainingMonitor]
 
+    def _get_sampling_scheduler(self):
+        """
+        Computes the threshold over which we train against model samples rather than gold targets
+        """
+        i = mx.sym.Variable('updates')
+
+        check_scheduled_sampling_params(self.config.scheduled_sampling_type, self.config.scheduled_sampling_params)
+
+        if self.config.scheduled_sampling_type == 'inv-sigmoid-decay':
+            k, lower_bound = self.config.scheduled_sampling_params
+            sampling_scheduler = lambda: k / (k + mx.sym.exp(i / k) - 1) * (1 - lower_bound) + lower_bound
```

Comment:
If lower_bound is just to lowerbound, wouldn't mx.sym.max() be enough?
Why doesn't exponential decay need it?